src/Microsoft.OData.Core/JsonLight/ODataJsonLightOutputContext.cs

```diff
@@ -889,6 +900,38 @@ protected override void Dispose(bool disposing)
             return jsonWriter;
         }
 
+        /// <summary>
+        /// Creates a new JSON writer of <see cref="IJsonWriter"/> that can write
+        /// directly to the output stream. Returns null if unable to create
+        /// such a writer with the given constraints.
+        /// </summary>
+        /// <param name="container">The dependency injection container to get related services like <see cref="IStreamBasedJsonWriterFactory"/>.</param>
+        /// <param name="outputStream">The output stream to write to.</param>
+        /// <param name="isIeee754Compatible">True if the writer should writer decimals and longs as strings.</param>
+        /// <param name="encoding">The text encoding of the output data.</param>
+        /// <returns>The JSON writer instance, or null if none could be created.</returns>
+        private static IJsonWriter CreateJsonWriter(
+            IServiceProvider container,
+            Stream outputStream,
+            bool isIeee754Compatible,
+            Encoding encoding)
+        {
+            if (container == null)
+            {
+                return null;
+            }
+
+            IStreamBasedJsonWriterFactory factory = container.GetService<IStreamBasedJsonWriterFactory>();
+            if (factory == null)
+            {
+                return null;

```

Comment:
I think we should default to the `DefaultStreamBasedJsonWriterFactory` instead here?

Reply:
The intention is to make this an "opt-in" feature for now, since it's still new. So if someone doesn't explicitly opt in by injecting the new factory, it will just fall back to the existing implementation. This avoids any surprises for customers who are unaware of this feature. And we can get feedback from customers who opt in to it.

Reply:
Are we ever expecting to make this the default for 7.x?

Reply:
To be honest, I'm not sure at this point. Maybe if get stable positive feedback from initial adopters and also close any compatibility gaps between this and the existing JsonWriter, then could consider making it default.